azurerm plugin crash after change to Front Door configuration

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.0.11
on linux_amd64
+ provider registry.terraform.io/hashicorp/azuread v2.7.0
+ provider registry.terraform.io/hashicorp/azurerm v2.86.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_frontdoor_rules_engine`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
  routing_rule {
    name = "disable-download-cache"
    accepted_protocols = ["Https", "Http"]
    patterns_to_match = ["/documents/download/*"]
    frontend_endpoints = ["www-frontend-ep"]
    forwarding_configuration {
      backend_pool_name   = "static-website"
      forwarding_protocol = "MatchRequest"
      cache_enabled = false

    }
  }
```

### Actual Behaviour

azurerm_frontdoor_rules_engine.disable_download_cache: Creating...
╷
│ Error: Plugin did not respond
│
│   with azurerm_frontdoor_rules_engine.disable_download_cache,
│   on fd.tf line 175, in resource "azurerm_frontdoor_rules_engine" "disable_download_cache":
│  175: resource "azurerm_frontdoor_rules_engine" "disable_download_cache" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v2.86.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 38 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.expandFrontDoorRulesEngineAction({0xc0004df840, 0x0, 0x5df6763})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_rules_engine_rule.go:283 +0x165
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.expandFrontDoorRulesEngineRules({0xc0004df7e0, 0x1, 0xc0006ea718})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_rules_engine_rule.go:339 +0x225
github.com/hashicorp/terraform-provider-azurerm/internal/services/frontdoor.resourceFrontDoorRulesEngineCreateUpdate(0x0, {0x550c8c0, 0xc000ba2e00})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/frontdoor/frontdoor_rules_engine_rule.go:262 +0x319
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc001bb9260, {0x659ef88, 0xc001bac300}, 0x2, {0x550c8c0, 0xc000ba2e00})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000db7880, {0x659ef88, 0xc001bac300}, 0xc001bb0750, 0xc0001a6680, {0x550c8c0, 0xc000ba2e00})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000d7a0, {0x659ef88, 0xc001bac300}, 0xc001d834a0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00135e560, {0x659f030, 0xc001e04300}, 0x4173eb)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:332 +0x6c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5a30d20, 0xc00135e560}, {0x659f030, 0xc001e04300}, 0xc001b15740, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000220a80, {0x65edb60, 0xc0005d1500}, 0xc001babd40, 0xc001360ff0, 0xaadd5c0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc000220a80, {0x65edb60, 0xc0005d1500}, 0xc001babd40, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v2.86.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Steps to Reproduce

1. simple `terraform apply` after new routing_rule was added to the code


* #0000

--- a/internal/services/frontdoor/frontdoor_rules_engine_rule.go
+++ b/internal/services/frontdoor/frontdoor_rules_engine_rule.go
@@ -276,7 +276,7 @@ func resourceFrontDoorRulesEngineCreateUpdate(d *pluginsdk.ResourceData, meta in
 }
 
 func expandFrontDoorRulesEngineAction(input []interface{}) frontdoors.RulesEngineAction {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return frontdoors.RulesEngineAction{}
 	}
 
@@ -294,7 +294,7 @@ func expandFrontDoorRulesEngineAction(input []interface{}) frontdoors.RulesEngin
 }
 
 func expandHeaderAction(input []interface{}) *[]frontdoors.HeaderAction {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 	output := make([]frontdoors.HeaderAction, 0)
@@ -319,7 +319,7 @@ func expandHeaderAction(input []interface{}) *[]frontdoors.HeaderAction {
 }
 
 func expandFrontDoorRulesEngineRules(input []interface{}) *[]frontdoors.RulesEngineRule {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 
@@ -346,7 +346,7 @@ func expandFrontDoorRulesEngineRules(input []interface{}) *[]frontdoors.RulesEng
 }
 
 func expandFrontDoorRulesEngineMatchCondition(input []interface{}) *[]frontdoors.RulesEngineMatchCondition {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 
@@ -381,7 +381,7 @@ func expandFrontDoorRulesEngineMatchCondition(input []interface{}) *[]frontdoors
 }
 
 func expandFrontDoorRulesEngineMatchConditionTransform(input []interface{}) *[]frontdoors.Transform {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return &[]frontdoors.Transform{}
 	}
 

